src/frontends/sql/decode.cc

```diff
@@ -49,6 +49,34 @@ static Value DecodeLiteral(const Literal &literal,
                       DecoderContext::kDefaultOutputName));
 }
 
+static Value DecodeMergeOperation(const MergeOperation &merge_operation,
+                                  DecoderContext &decoder_context) {
+  // Although the proto separates the inputs and the control inputs, we can't
+  // really take advantage of that yet. We do, however, expect that at least one
+  // regular input is provided.
+  CHECK(merge_operation.inputs_size() >= 1)
+      << "Each MergeOperation is expected to have at least one non-control "
+         "input.";
+  std::vector<Value> direct_input_values;
+  direct_input_values.reserve(merge_operation.inputs_size());
+  for (Expression input_expr : merge_operation.inputs()) {
+    direct_input_values.push_back(
+        DecodeExpression(input_expr, decoder_context));
+  }
```

Comment:
I don't think that will work here, because `inputs` does not return a `std::vector`. It instead returns a `RepeatedPtrField`. But I will keep this utility in mind for the future.